### PR TITLE
MDCT-1857 pt. 1 -- add form table and api

### PR DESCRIPTION
**Pull request by gmrabian**
## Description
Partial [MDCT-1857](https://qmacbis.atlassian.net/browse/MDCT-1857) Adds the backend table and API by which we will fetch exact forms or push forms.

### How to test

Ask me for help if this is confusing

There is no frontend component yet so you can test against the API directly
- `./dev local`
- View the Network tab on your browser. Find an API request (like banner or dashboard) and get the `x-api-key` security header value. Put that key: value in your preferred API testing tool (I use Postman)
- `POST localhost:3030/local/forms` to create a form (needs `formId` and `formTemplate` in the body)
- `GET localhost:3030/local/forms/{formId}` to test getting a specific form, using the formId you created

### Changed Dependencies
**Add one line to your root .env file (it's in the README for reference)**

## Code author checklist
- [x] I have performed a self-review of my code
- [x] I have added [thorough](https://bit.ly/3zPrxuZ) tests
- [x] I have added analytics, if necessary
- [x] I have updated the documentation, if necessary

## Reviewer checklist (two different people)
- [ ] I have done the deep review and verified the items in the above checklist are g2g
- [ ] I have done the lgtm review

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -64,6 +64,7 @@ URL=http://localhost/3000
 BANNER_TABLE_NAME=local-banners
 REPORT_DATA_TABLE_NAME=local-mcpar-reports-data
 REPORT_TABLE_NAME=local-mcpar-reports
+FORM_TABLE_NAME=local-forms
```
Comment (braxex) [inline README.md:67]: i'm guessing we'll want to keep `mcpar` separate from `mlr` and `naaar`, so maybe the name should be `local-mcpar-forms`?

File: services/app-api/handlers/forms/get.test.ts [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+jest.mock("../../utils/dynamo/dynamodb-lib", () => ({
+  __esModule: true,
+  default: {
+    query: jest.fn().mockReturnValue({
+      Items: [
+        {
+          createdAt: 1654198665696,
+          formName: "TEST",
+          formId: "TEST_test-1-22",
+          formJson: {},
```
Comment (braxex) [inline services/app-api/handlers/forms/get.test.ts:16]: the thing that gets stored here will be the entire `mcpar.json` file, + validation from mcpar/index as we talked about.

File: services/app-api/handlers/forms/get.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+  const latestForm = scannedResults.filter(
+    (item) => parseInt(item.createdAt) === latestFormTimestamp
+  );
+
+  return {
+    status: StatusCodes.SUCCESS,
+    body: latestForm[0],
+  };
+});
```
Comment (braxex) [inline services/app-api/handlers/forms/get.ts:60]: i might be thinking about this wrong, but i don't think we'll ever need to get the latest form. the latest form will always be the form defined in the code, and when we create a new report on the dashboard, we can just use that 'local' form as the form that gets associated to the new report. then when we pull down reportData (on entry into form) we also pull down the associated form and populate the entire report fill-out and validation experience from that. so we wouldn't ever need the latest form -- just the form as specified by name/id so we could pull it down on a report-by-report basis.

File: services/app-api/serverless.yml [hunk 4]
```diff
@@ -210,6 +212,27 @@ functions:
...
+                formName: true
+                formId: true
+  getLatestForm:
+    handler: handlers/forms/get.getLatestForm
+    events:
+      - http:
+          path: forms
+          method: get
+          cors: true
+          authorizer: ${self:custom.authValue.${self:custom.stage}, ""}
```
Comment (braxex) [inline services/app-api/serverless.yml:235]: see note above.

**Review by braxex — COMMENTED** [review]
Just added some thoughts. Let's chat about these if you disagree so I can understand better why I'm wrong. 😄

File: services/app-api/handlers/forms/get.test.ts (hunk 2 above)
Comment (gmrabian) [reply services/app-api/handlers/forms/get.test.ts:16]: Makes sense

File: README.md (hunk 1 above)
Comment (gmrabian) [reply README.md]: updated

File: services/app-api/handlers/forms/get.ts (hunk 3 above)
Comment (gmrabian) [reply services/app-api/handlers/forms/get.ts]: removed

**Review by braxex — APPROVED** [review]
All looks good to me.

**Pull request merged**